addons/mail/static/src/models/thread/thread.js

```diff
@@ -983,6 +990,16 @@ function factory(dependencies) {
             ]];
         }
 
+        /**
+         *
+         * @private
+         */
+        _computePendingSeenMessageId() {
+            if (this.pendingSeenMessageId) {
+                return this.pendingSeenMessageId;
+            }
+            return this.seen_message_id;
+        }
```

Comment:
Don't do this (cf previous comments to check both variables). If you need a state that is the merge of the two, need a 3rd field. (see fold state)